dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -332,7 +319,98 @@ public String toJson(Contentlet con) throws DotMappingException {
 		}
 	}
 
-    /**
+	private void generateBinaryMetadata(final Contentlet contentlet,
+										final StringWriter stringWriter,
+										final Map<String, Object> mapLowered,
+										final Field field) throws Exception {
+
+		//Verify if it is enabled the option to regenerate missing metadata files on reindex
+		final boolean regenerateMissingMetadata = Config
+				.getBooleanProperty("regenerate.missing.metadata.on.reindex", true);
+                /*
+                Verify if it is enabled the option to always regenerate metadata files on reindex,
+                enabling this could affect greatly the performance of a reindex process.
+                 */
+		final boolean alwaysRegenerateMetadata = Config
+				.getBooleanProperty("always.regenerate.metadata.on.reindex", false);
+
+		Map<String, Object> metadataMap = null;
+
+		if (contentlet.isLive() || contentlet.isWorking()) {
+
+			final Optional<com.dotcms.contenttype.model.field.FieldVariable> customIndexMetaDataFieldsOpt =
+					FactoryLocator.getFieldFactory().byFieldVariableKey(field.getIdentifier(), BinaryField.INDEX_METADATA_FIELDS);
+			final TikaUtils tikaUtils = new TikaUtils();
+
+			final Set<String> metadataFields = customIndexMetaDataFieldsOpt.isPresent()?
+					new HashSet<>(Arrays.asList(customIndexMetaDataFieldsOpt.get().value().split(StringPool.COMMA))):
+					tikaUtils.getConfiguredMetadataFields();
+
+			final File binaryField = contentlet.getBinary(field.getVelocityVarName());
+			if (null != binaryField && binaryField.exists() && binaryField.canRead()) {
+
+				if (alwaysRegenerateMetadata) {
+					metadataMap = new TikaUtils().generateMetaDataForce(contentlet, binaryField, metadataFields);
+				} else if (regenerateMissingMetadata) {
+					metadataMap = new TikaUtils().generateMetaData(contentlet, binaryField, metadataFields);
+				}
+			}
+
+			// see if we have content metadata
+			if(null != metadataMap) {
+
+				for (final String metadataKey : metadataFields) {
+
+					final Object metadataValue = metadataMap.get(metadataKey);
+					mapLowered.put(FileAssetAPI.META_DATA_FIELD.toLowerCase() + StringPool.PERIOD + metadataKey, metadataValue);
+					stringWriter.append(metadataValue.toString()).append(' ');
+				}
+			}
+		}
+	}
+
+	private Optional<Field> findFirstBinaryFieldIndexable(final Contentlet contentlet) {
+
+		final List<Field> fields  = new ArrayList<>(
+				FieldsCache.getFieldsByStructureInode(contentlet.getStructureInode()));
+
+		return fields.stream().filter(Field::isIndexed)
+				.filter(field -> field.getFieldType().equals(Field.FieldType.BINARY.toString()))
+				.findFirst();
+	}
+
+	private void generateFileAssetMetadata(final Contentlet contentlet,
+										   final StringWriter stringWriter,
+										   final Map<String, Object> mapLowered) throws Exception {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)